plugins/github-dev/skills/cr-fix/tests/run-tests.sh

```
@@ -586,6 +586,37 @@ gcap2=$(bash -euo pipefail -c '
+is "in-repo nested file"             "$(pt 'sub/nested/deep.txt')"  0
+# The whole reason `-m` was there: a fix may create a file that does not exist yet.
+is "not-yet-created file in repo"    "$(pt 'sub/brand-new.md')"     0
+is "not-yet-created nested dir file" "$(pt 'sub/nested/new.txt')"   0
+# Containment must still hold.
+is "parent escape rejected"          "$(pt '../escape.md')"         1
+is "mid-path .. rejected"            "$(pt 'sub/../../escape.md')"  1
+is "absolute path rejected"          "$(pt '/etc/passwd')"          1
+is "home-relative rejected"          "$(pt '~/secrets')"            1
+# A symlink pointing outside the repo must not smuggle a write past the gate.
+ln -s "$PT_ROOT/../pt-outside-$$" "$PT_ROOT/link-out" 2>/dev/null \
+  && is "symlink escaping repo rejected" "$(pt 'link-out/x.md')" 1 \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)